src/containers/BookList.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import Book from '../components/Book'
+import CategoryFilter from '../components/CategoryFilter'
+import { connect } from 'react-redux'
+
+const BookList = ({ books, filter }) => {
+  books = filter === 'All' ? books : books.filter(book => book.category === filter)
```

Comment:
A neater way to do this is filter everything in the state level. 
```javascript
const mapStateToProps = state => ({
  books: state.filter === 'All' ? true : state.books.filter(({category}) => category === state.filter)
})
```
To avoid the over-population of the state here we can write a 'filtering' function in another file (which we import here):

```javascript
const filterBooks = (books, categoryFilter) => {
 return books.filter(({category}) => categoryFilter === "All" ? true : category === categoryFilter
}
```

Which we can call in mapStateToProps:

```javascript
books: filterBooks(state.books, state.filter)
```